src/main/java/io/cdap/plugin/gcp/spanner/common/SpannerUtil.java

```diff
@@ -47,21 +49,26 @@ public static Spanner getSpannerService(String serviceAccountFilePath, String pr
   /**
    * Validate that the schema is a supported one, compatible with Spanner.
    */
-  public static void validateSchema(Schema schema, Set<Schema.Type> supportedTypes) {
+  public static void validateSchema(Schema schema, Set<Schema.Type> supportedTypes, FailureCollector collector) {
     for (Schema.Field field : schema.getFields()) {
       Schema fieldSchema = field.getSchema();
       fieldSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable() : fieldSchema;
 
       Schema.LogicalType logicalType = fieldSchema.getLogicalType();
       if (logicalType != null && !SUPPORTED_LOGICAL_TYPES.contains(logicalType)) {
-        throw new IllegalArgumentException(String.format("Schema logical type %s not supported.",
-                                                         logicalType));
+        collector.addFailure(
+          String.format("Field schema of type '%s' is unsupported.", fieldSchema.getDisplayName()),
+          "Supported types are: date and timestamp").withOutputSchemaField(field.getName(), null);
       }
 
       if (logicalType == null) {
         Schema.Type type = fieldSchema.getType();
         if (!supportedTypes.contains(type)) {
-          throw new IllegalArgumentException(String.format("Schema type %s not supported.", type));
+          collector.addFailure(
+            String.format("Field schema of type '%s' is unsupported.", fieldSchema.getDisplayName()),
+            String.format("Supported types are: %s",
+                          supportedTypes.stream().map(t -> t.name().toLowerCase()).collect(Collectors.joining(","))))
```

Comment:
display name?

Reply:
I kept it as is because these types does not include logical types. because of this check `logicalType == null`